ui/packages/app/web/src/index.tsx

```diff
@@ -11,16 +11,34 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import {tryLoadAndStartRecorder} from '@alwaysmeticulous/recorder-loader';
 import {createRoot} from 'react-dom/client';
 
 import App from './App';
 import reportWebVitals from './reportWebVitals';
 
-const container = document.getElementById('root');
-const root = createRoot(container!);
-root.render(<App />);
+const shouldEnableMeticulousRecorder =
+  process.env.NODE_ENV === 'development' || process.env.VERCEL_ENV === 'preview';
```

Comment:
Should the `isDevMode` utility be used?

https://github.com/parca-dev/parca/blob/df4fafb8a6639867aae35f1790b2df4e0d279d00/ui/packages/shared/utilities/src/index.ts#L96-L98

It be nice to add `|| process.env.VERCEL_ENV === 'preview'` to it as well, it'd fix the path prefix in the preview :)

Reply:
I think largely we want previews to be as similar to prod deployments as possible though, no? So changing is dev might cause this to blur further.

Reply:
Agreed. Modifying the `isDevMode` utility would change how that behaves .

Reply:
It would not change much in behavior, the function is only used here:

https://github.com/parca-dev/parca/blob/830413a74b8d89f5aa9f35ba20fe3d4b4837b883/ui/packages/app/web/src/App.tsx#L76-L78

Maybe rename it to `isDevModeOrPreview`?

The idea would be to fix this kind of links in the preview:
https://parca-hn9um2wya-parca-dev.vercel.app/%7B%7B.PathPrefix%7D%7D/targets

Double check if it'd be the fix tho, something is off in the prefix handling for dev/preview, but I understand very little about React :smile:

Reply:
Ahh I'm understanding you now. Had no idea that was happening, I'll try it out to see if it's a fix